applications/hpcc/hpcc_random_access.cpp

```diff
@@ -0,0 +1,120 @@
+// to run, do something like
+// make -j demo-hpcc_random_access
+// bin/grappa_run --ppn 8 --nnode 12 -- demo-hpcc_random_access.exe 
+
+#include <Grappa.hpp>
+#include <iomanip>
+
+using namespace Grappa;
+
+// define command-line flags (third-party 'gflags' library)
+DEFINE_int64( scale, 20, "table size = 2 ^ scale * sizeof(uint64_t) * number of PEs" );
+DEFINE_int64( iters, 4, "Number of iterations" );
+
+// define custom statistics which are logged by the runtime
+// (here we're not using these features, just printing them ourselves)
+GRAPPA_DEFINE_METRIC( SimpleMetric<double>, gups_runtime, 0.0 );
+GRAPPA_DEFINE_METRIC( SimpleMetric<double>, gups_throughput, 0.0 );
+
+#define POLY 0x0000000000000007ULL
+#define PERIOD 1317624576693539401LL
+
+typedef double time_type;
+std::string print_time(time_type t)
+{
+	std::ostringstream out;
+	out << std::setiosflags(std::ios::fixed) << std::setprecision(2) << t;
+	return out.str();
+}
+
+
+uint64_t N;
+uint64_t HPCC_starts(int64_t n);
+
+Grappa::GlobalCompletionEvent randomaccess_gce;
+
+template <Grappa::GlobalCompletionEvent * GCE = &randomaccess_gce >
```

Comment:
You wouldn't necessarily need to make this a template parameter for `run_random_access`. Is there a reason to not just use `randomaccess_gce` directly? (or for that matter, to not use the default gce?